lib/src/facts/aix/load_average_resolver.cc

```diff
@@ -0,0 +1,50 @@
+#include <internal/facts/aix/load_average_resolver.hpp>
+#include <leatherman/logging/logging.hpp>
+#include <leatherman/locale/locale.hpp>
+
+#include <sys/inttypes.h>
+#include <sys/kinfo.h>
+
+// Mark string for translation (alias for leatherman::locale::format)
+using leatherman::locale::_;
+
+using namespace std;
+
+/**
+ * This system call lets us query the kernel directly for system
+ * information. We use it to get our current load averages.
+ *
+ * @param info the info we're retrieving from the kernel.
+ * @param buf the buffer that we'll store the information in
+ * @param buf_size a pointer to the variable containing the size of the buffer in bytes
+ * @param arg no idea what this param. represents. we will usually set this to 0.
+ * @return 0 if we successfully retrieve the information, else a negative value
+ */
+extern "C" int getkerninfo(int info, char* buf, int* buf_size, int32long64_t arg);
+
+// Converts the given integer average into a load average.
+static double to_load_avg(double average) {
+  // 65536 is the load average scale on AIX machines.
+  return average / 65536;
```

Comment:
I have no idea how the screen source code figured out that `65536` is the load scale. I've grepped through all of the include files on my AIX machine and couldn't find this declaration. IBM's docs don't point to anything either. I don't know how often this will change from AIX machine to AIX machine, but if it does change it might be worthwhile to try and get this info. from the `uptime` command instead.

For what's it's worth, the screen code defines it like this (in their configure.ac file).
```
# if defined(apollo) || defined(_IBMR2) || defined(_AUX_SOURCE) || defined(AUX)
loadscale=65536
```

Reply:
I could take a crack at disassembling the `uptime` command. Maybe I'll try that.

Reply:
Nope. The disassemble doesn't really give any useful info.